### PR TITLE
Add Renko indicators for issue #1187

### PR DESCRIPTION
**Pull request by TheCookieLab**
Changes proposed in this pull request:
- Add `RenkoUpIndicator`, `RenkoDownIndicator`, and `RenkoXIndicator` with a shared `RenkoCounter` helper to compute brick state
- Add comprehensive Renko indicator tests that cover reversals, exact brick boundaries, and multi-brick price moves
- Document Renko package guidance, record headless `ta4j-examples` testing tips, and update the changelog for #1187

- [X] added an entry with related ticket number(s) to the unreleased section of `CHANGES.md`

------
https://chatgpt.com/codex/tasks/task_e_68e58533a34483269f8285bf3f889df5

### CODE REVIEW

File: ta4j-core/src/main/java/org/ta4j/core/indicators/renko/RenkoCounter.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        if (index < 0) {
+            throw new IllegalArgumentException("index must be non-negative");
+        }
+        var state = cache.get(index);
+        if (state != null) {
+            return state;
+        }
+        var calculated = calculateState(index);
+        cache.put(index, calculated);
+        return calculated;
```
[the remark this answers is not in the document]
Comment (TheCookieLab) [reply ta4j-core/src/main/java/org/ta4j/core/indicators/renko/RenkoCounter.java:66]: 👍
